packages/firebase_data_connect/firebase_data_connect/lib/src/core/ref.dart

```diff
@@ -0,0 +1,82 @@
+part of firebase_data_connect;
+
+abstract class Listener<Data, Variables> {
+  Stream<OperationResult<Data, Variables>> onResult();
+  // void onError();
+}
+
+class QueryManager {
+  Map<String, Map<String, StreamController<dynamic>>> trackedQueries = {};
+  Stream addQuery<Variables>(
+      String queryName, Variables variables, String varsAsStr) {
+    // TODO(mtewani): Replace with more stable encoder
+    String key = varsAsStr;
+    if (trackedQueries[queryName] == null) {
+      trackedQueries[queryName] = <String, StreamController>{};
+      print("Creating map for $queryName");
+      debugPrint('Creating map for $queryName');
+    }
+    if (trackedQueries[queryName]![key] == null) {
+      print("Creating stream for $queryName");
```

Comment:
Don't invoke 'print' in production code.

Reply:
Done